### PR TITLE
[v3.0] Use cachix for static build & include podman-remote build

### PR DESCRIPTION
**Pull request by cevich**
Backport from master

### CODE REVIEW

**Comment by cevich** [comment]
@saschagrunert PTAL...Does the cache "key" need to be altered (on this branch) somehow, so the cache is distinct from the master branch?  *note* we have the `$DEST_BRANCH` variable available for use in `.cirrus.yml` and every other script, in case that helps.

**Comment by saschagrunert** [comment]
> @saschagrunert PTAL...Does the cache "key" need to be altered (on this branch) somehow, so the cache is distinct from the master branch? _note_ we have the `$DEST_BRANCH` variable available for use in `.cirrus.yml` and every other script, in case that helps.

This looks fine from my side. The cache can be re-used, even if some derivations differ. Cachix will just put both into the same cache.

File: .cirrus.yml [hunk 1]
```diff
@@ -335,7 +335,7 @@ static_alt_build_task:
         - build
     # Community-maintained task, may fail on occasion.  If so, uncomment
     # the next line and file an issue with details about the failure.
-    allow_failures: $CI == $CI
+    # allow_failures: $CI == $CI
```
Comment (saschagrunert) [inline .cirrus.yml:338]: Do we want to disallow failures now?

**Review by saschagrunert — APPROVED** [review]

File: .cirrus.yml (hunk 1 above)
Comment (cevich) [reply .cirrus.yml:338]: I *think* so.  I did this on master as well.  The switch to cachix seems to have greatly improved the reliability of this job.  I'm puzzled why Cirrus's cache failed us...but "proof of the pudding is in the eating" :shrug:

**Comment by cevich** [comment]
> Cachix will just put both into the same cache side by side with a different checksum.

Okay, glad to hear.  This is what I wasn't sure about.

**Pull request merged**